pkg/apply/applier.go

```diff
@@ -50,6 +50,8 @@ func NewApplier(factory util.Factory, ioStreams genericclioptions.IOStreams) *Ap
 	a.previousInventoriesFunc = a.previousInventories
 	a.infoHelperFactoryFunc = a.infoHelperFactory
 	a.manifestReaderFunc = a.readObjects
+	a.InventoryFactoryFunc = prune.WrapInventoryObj
```

Comment:
I don't think we could set a property, since the logic for creating the type might not live in this repository. Let's discuss this in person. If we move in another direction, I think this would be the best first implementation since it is a small amount of code. Whatever we implement, we can not import the overriden type into this code base.